### ISSUE TITLE
runScanner does not always return final state

### ISSUE THREAD

**Issue by wereHamster** [bos/attoparsec#105]
Here's a bit of code that I expected to extract up to four lowercase characters from the input string. But the result is an empty string (`Right ""`).

```haskell
runScannerTest :: Either String Text
runScannerTest = AP.parseOnly p "lowER"
  where
    p = do
        (s, firstFourLowercaseLetters) <- AP.runScanner [] f
        return $ T.pack $ reverse firstFourLowercaseLetters

    f :: String -> Char -> Maybe String
    f acc c = if length acc < 4 && isLower c
        then Just $ c:acc
        else Nothing
```

If I'm reading the implementation of `runScanner` correctly, then there is at least one case where it returns the initial state instead of the last state: `scan_` -> `go` -> `scanner` which returns `Finished` and then `f s (T.take n input : acc)` where `s` == `s0` which is the initial state.

What's even more strange is that depending on how I feed the data to the parser I get different results:

```haskell
AP.eitherResult $ AP.feed (AP.parse p "lo") "wER"
```

returns `Right "lo"`

**Comment by tolysz** [comment]
Hi @wereHamster your report was so good that I could make the fix in minutes ;) especially that I bumped into this issue myself ;)

**Comment by tolysz** [comment]
@bgamari Any chance to bump version and publish to hackage the updated code?

**Comment by bgamari** [comment]
Ahh, yes. I don't see why not; sorry this slipped!

**Comment by bgamari** [comment]
Released as 0.13.1.0.

**Comment by tolysz** [comment]
Thakns :)
